src/System.Text.Primitives/System/Text/Encoding/Utf16/Utf16TextEncodingLE.cs

```diff
@@ -7,113 +7,145 @@ namespace System.Text.Utf16
 {
     class Utf16TextEncodingLE : TextEncoder
     {
-        public override bool TryEncodeFromUtf8(ReadOnlySpan<byte> utf8, Span<byte> buffer, out int bytesWritten)
+        #region Encoding Constants
+
+        // TODO: Some of these members are needed only in Utf16LittleEndianEncoder.
+        //       Should we add the usage of them to UnicodeCodePoint class and merge this class with it?
+        private const uint Utf16HighSurrogateFirstCodePoint = 0xD800;
```

Comment:
There was a comment left in UnicodeConstants that it might be useful to merge with the UTF-16 encoder.

I thought about this for a bit and came to the conclusion that only the UTF-16 encoder needs these values. The other copy currently exists in an old / existing class that will eventually be removed (and is currently hidden from the surface).